### DISCUSSION TITLE
Startup Task

### DISCUSSION THREAD

**Discussion by detroitpro** [TypedProject/ts-express-decorators#1509] [Q&A] [answered]
Is it suggested to use [Agenda](https://tsed.io/tutorials/agenda.html) for a `task` that needs to run on startup? If so, where can I find an example.

I need to check if a record has been created in the database and create it if not.

**Comment by detroitpro** [comment, marked as answer]
I think using `hooks` is the answer. 
```typescript
export class Server {
  @Inject()
  app: PlatformApplication | undefined;

  @Inject() 
  workers: Workers | undefined; // inject `service` here.

  @Configuration()
  settings: Configuration | undefined;

  $beforeRoutesInit(): void {
    this.app
      ?.use(cors())
      .use(cookieParser())
      .use(compress({}))
      .use(methodOverride())
      .use(express.json())
      .use(express.urlencoded({ extended: true }));
  }

  async $onReady(): Promise<void> { // use $onReady hook to perform action when server is started
    await this.workers?.init();
  }
}
```

**Reply by Romakita** [reply]
I don't understand the relation with Agenda for your needs ?

**Reply by Romakita** [reply]
But yes, the $onReady hook is the right way to check something when the server is up. But you don't need to intercept the hook on the server. Any service/provider can intercept a hook. So you can implement the $onReady on the Wokers service directly.

```typescript
@Injectable()
class Workers {
  init() {
  }
  $onReady() {
     this.init()
  }
}
```

See you
Romain
